### PR TITLE
outbuffer.d: add overflow checks

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Comment by WalterBright** [comment]
This uncovered a latent bug - some platforms require a `va_list, va_copy, va_end`.

File: std/outbuffer.d [hunk 1]
```diff
@@ -269,21 +272,37 @@ class OutBuffer
...
             {
-                count = vsnprintf(p,psize,f,args);
+                va_list args2;
+                va_copy(args2, args);
+                count = vsnprintf(p,psize,f,args2);
+                va_end(args2);
                 if (count != -1)
                     break;
-                psize *= 2;
+
```
Comment (andralex) [inline std/outbuffer.d]: Another overkill change. If you really insist:

`psize <= psize.max / 2 || assert(0);`

File: std/outbuffer.d [hunk 2]
```diff
@@ -269,21 +272,37 @@ class OutBuffer
...
             }
             else version(Posix)
             {
-                count = vsnprintf(p,psize,f,args);
+                import core.checkedint : addu, mulu;
+                bool overflow;
+
+                va_list args2;
+                va_copy(args2, args);
+                count = vsnprintf(p,psize,f,args2);
```
Comment (andralex) [inline std/outbuffer.d]: space after comma

**Comment by andralex** [comment]
grudgingly fine after fixing the spacing

File: std/outbuffer.d (hunk 2 above)
Comment (WalterBright) [inline std/outbuffer.d]: done

File: std/outbuffer.d (hunk 1 above)
Comment (WalterBright) [inline std/outbuffer.d]: done

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Pull request merged**